### PR TITLE
Implement simple registration code feature.

### PR DESCRIPTION
**Pull request by mlen108**

### CODE REVIEW

**Comment by mlen108** [comment]
@maxpeterson can you review this?

File: profiles/forms.py [hunk 1]
```diff
@@ -37,3 +42,16 @@ def __init__(self, *args, **kwargs):
                 self.fields[fname].required = True
             except KeyError:
                 pass
+
+
+class CodeField(fields.CharField):
```
Comment (maxpeterson) [inline profiles/forms.py]: Can this be moved to profiles/modules/codes/fields.py

File: profiles/modules/codes/extensions/codes.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
+
+def register(cls, admin_cls):
+    cls.add_to_class('codes', models.ManyToManyField('profiles.Code',
```
Comment (maxpeterson) [inline profiles/modules/codes/extensions/codes.py]: I was expecting a ForeignKey.

File: profiles/forms.py (hunk 1 above)
Comment (mlen108) [inline profiles/forms.py]: Fixed in 09b3b9b

File: profiles/modules/codes/fields.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    re_clean_code = re.compile('[^a-zA-Z0-9]+')
+
+    def validate(self, value):
+        super(CodeField, self).validate(value)
+
+        value = self.re_clean_code.sub('', value)
+        try:
+            Code.objects.get(code=value)
+        except Code.DoesNotExist:
+            raise exceptions.ValidationError('Enter a valid registration code.')
```
Comment (maxpeterson) [inline profiles/modules/codes/fields.py]: I think it needs a to_python like ModelChoiceField

```
def to_python(self, value):
    try:
        key = self.to_field_name or 'code'
        value = Code.objects.get(**{key: value})
    except (ValueError, Code.DoesNotExist):
        raise ValidationError('Enter a valid registration code.')
    return value
```

File: profiles/modules/codes/models.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
+from django.db import models
+from django.utils.safestring import mark_safe
+
+class Code(models.Model):
+    """
+    Registration codes
+    """
+    code = models.CharField(max_length=8)
```
Comment (maxpeterson) [inline profiles/modules/codes/models.py]: Make the max_length 255

**Pull request merged**